azure-kusto-data/azure/kusto/data/security.py

```diff
@@ -47,8 +47,10 @@ def _get_azure_cli_auth_token() -> dict:
 
             # TODO: not sure I should take the first
             return data[0]
-        except Exception:
-            pass
+        except Exception as e:
+            raise KustoAuthenticationError(
+                AuthenticationMethod.az_cli_profile, e, message="Azure cli token was not found. Please run 'az login' to setup account."
+            )
```

Comment:
self.kusto_uri is the resource.
I know the exception is resource agnostic, but you can still fill it in.
You can also pass a mock authority, as done with MSI & Token Provider authentication

Reply:
This function is a stand alone, no class context, and I don't think passing this as an argument worth it.

Reply:
You can throw KustoClientError instead and use the add the missing details in acquire_authorization_header()
The resource is already set, just add an authority as done for MSI and Token Provider